test-templates/src/glv.rs

```diff
@@ -0,0 +1,65 @@
+use std::ops::Mul;
+
+use ark_ec::{
+    scalar_mul::glv::GLVConfig,
+    short_weierstrass::{Affine, Projective},
+    AffineRepr, CurveGroup, Group,
+};
+use ark_std::UniformRand;
+
+pub fn glv_scalar_decomposition<P: GLVConfig>() {
+    let mut rng = ark_std::test_rng();
+    for _i in 0..100 {
+        let k = P::ScalarField::rand(&mut rng);
+
+        let (k1, is_k1_positive, k2, is_k2_positive) = <P as GLVConfig>::scalar_decomposition(k);
+
+        if is_k1_positive && is_k2_positive {
+            assert_eq!(k1 + k2 * P::LAMBDA, k);
+        }
+        if is_k1_positive && !is_k2_positive {
+            assert_eq!(k1 - k2 * P::LAMBDA, k);
+        }
+        if !is_k1_positive && is_k2_positive {
+            assert_eq!(-k1 + k2 * P::LAMBDA, k);
+        }
+        if !is_k1_positive && !is_k2_positive {
+            assert_eq!(-k1 - k2 * P::LAMBDA, k);
+        }
+        // could be nice to check if k1 and k2 are indeed small.
```

Comment:
I think we can do this via https://docs.rs/num-bigint/latest/num_bigint/struct.BigUint.html#method.bits